components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -682,12 +682,8 @@ class SitePermissionsFeature(
      * Re-attaches a fragment that is still visible but not linked to this feature anymore.
      */
     private fun reattachFragment(fragment: SitePermissionsDialogFragment) {
-        val session = sessionManager.findSessionById(fragment.sessionId)
-        val contentState = getCurrentContentState()
-
-        if (session == null || contentState == null ||
-            (noPermissionRequests(contentState))
-        ) {
+        val currentTab = store.state.findTabOrCustomTabOrSelectedTab(fragment.sessionId)?.content
```

Comment:
Here we need to warranty that when the provided `sessionId` is not found we have to fallback to selected tab.